src/Observer.ts

```diff
@@ -21,6 +21,68 @@ export interface CompletionObserver<T> {
 
 export type PartialObserver<T> = NextObserver<T> | ErrorObserver<T> | CompletionObserver<T>;
 
+// This class ObserverDoc is needed just so that ESDoc sees this documentation
+/**
+ * An interface for a consumer of push-based notifications delivered by an
+ * {@link Observable}.
+ *
+ * ```ts
+ * interface Observer<T> {
+ *   isUnsubscribed?: boolean;
+ *   next: (value: T) => void;
+ *   error: (err: any) => void;
+ *   complete: () => void;
+ * }
+ * ```
+ *
+ * An object conforming to the Observer interface is usually
+ * given to the `observable.subscribe(observer)` method, and the Observable will
+ * call the Observer's `next(value)` method to provide notifications. A
+ * well-behaved Observable will call an Observer's `complete()` method exactly
+ * once or the Observer's `error(err)` method exactly once, as the last
+ * notification delivered.
+ *
+ * @interface
+ * @name Observer
+ * @noimport true
+ */
+export class ObserverDoc<T> {
```

Comment:
interesting, does this build as part of the output? That wouldn't be ideal.

Too bad there isn't a way to document interfaces. Perhaps we add some files to the build that are purely for documenting interfaces and build them separately?